synapse/notifier.py

```diff
@@ -160,6 +154,7 @@ def on_new_room_event(self, event, extra_users=[]):
         listening to the room, and any listeners for the users in the
         `extra_users` param.
         """
+        yield run_on_reactor()
```

Comment:
Previously ``on_new_room_event`` hit the database, ``synapse.handlers.BaseHandler._create_new_client_event`` depended on that to make sure the notifications didn't block the request that created a new request. Since we no longer hit the database we need another way to cycle through the reactor before poking the notification_listeners.